### DISCUSSION TITLE
gcc postinstall fails with segmentation fault on Debian 10

### DISCUSSION THREAD

**Discussion by TuggyNE** [Homebrew/discussions#4390] [Linux] [answered]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 931327df1f30a3a5b1aa07657c0d1070e61988f9
Last commit: 7 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: faee70d0421d3f6ead8706a6b3d3eb3dc899fd4b
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 07 Apr 14:56 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: quad-core 64-bit amd_k8
Clang: N/A
Git: 2.20.1 => /bin/git
Curl: 7.64.0 => /bin/curl
Kernel: Linux 4.19.0-23-amd64 x86_64 GNU/Linux
OS: Debian GNU/Linux 10 (buster) (buster)
Host glibc: 2.28
/usr/bin/gcc: 8.3.0
/usr/bin/ruby: N/A
glibc: 2.35_1
gcc@11: N/A
gcc: 12.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I set up a Homebrew installation some time ago on this particular VM, and it seemed to be working, installing several packages (`brew list`: `binutils  fzf  gcc  gmp  isl  libmpc  mpfr  ncurses  zlib  zstd`). I came back to install a new package, and it all broke due to gcc. I saw `Warning: The post-install step did not complete successfully You can try again using: brew postinstall gcc` after installing `borkdude/brew/babashka` (which requires gcc, I guess like most Linuxbrew packages). `brew postinstall --debug gcc` shows a segmentation fault from gcc running `/home/linuxbrew/.linuxbrew/Cellar/gcc/12.2.0/bin/gcc-12 -print-libgcc-file-name` (obviously a smoke test, presumably for this very kind of situation).

I tried running `brew uninstall gcc`, `brew cleanup --purge=all -s` and `brew install gcc`, but that didn't help, so it wasn't some package corruption. `/usr/bin/gcc --version` runs fine (`gcc (Debian 8.3.0-6) 8.3.0`), although I don't know why it's so old. The instructions for bootstrapping Linuxbrew include apt package installs (`sudo apt-get install build-essential procps curl file git`) but those are all up to date. Reinstalling `build-essential` does nothing helpful.

**Comment by carlocab** [comment, marked as answer]
This sounds like https://github.com/Homebrew/homebrew-core/issues/126054, which is a bug in `patchelf.rb`.

**Reply by TuggyNE** [reply]
Yes; following the same steps (but for gcc@12) to revert Homebrew worked fine. Thanks!
